immutables-exceptions/src/main/java/com/hubspot/immutables/validation/ImmutableConditions.java

```diff
@@ -10,7 +12,7 @@ public static void checkValid(boolean expression, String template, Object... arg
     }
   }
 
-  public static <T> T checkNotNull(T ref, String template, Object... arguments) {
+  public static @Nonnull <T> T checkNotNull(T ref, String template, Object... arguments) {
```

Comment:
Can you please put the annotation on the previous line?
```suggestion
  @Nonnull
  public static <T> T checkNotNull(T ref, String template, Object... arguments) {
```